network/alsp/manager/manager_test.go

```diff
@@ -0,0 +1,721 @@
+package alspmgr_test
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/metrics"
+	mockmodule "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/alsp"
+	"github.com/onflow/flow-go/network/alsp/internal"
+	alspmgr "github.com/onflow/flow-go/network/alsp/manager"
+	mockalsp "github.com/onflow/flow-go/network/alsp/mock"
+	"github.com/onflow/flow-go/network/alsp/model"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/internal/testutils"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/p2p/conduit"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestHandleReportedMisbehavior tests the handling of reported misbehavior by the network.
+//
+// The test sets up a mock MisbehaviorReportManager and a conduitFactory with this manager.
+// It generates a single node network with the conduitFactory and starts it.
+// It then uses a mock engine to register a channel with the network.
+// It prepares a set of misbehavior reports and reports them to the conduit on the test channel.
+// The test ensures that the MisbehaviorReportManager receives and handles all reported misbehavior
+// without any duplicate reports and within a specified time.
+func TestHandleReportedMisbehavior(t *testing.T) {
```

Comment:
This seems to be testing the network layer correctly passes the report to the manager
```suggestion
func TestNetworkPassesReportedMisbehavior(t *testing.T) {
```